base_layer/core/src/transactions/transaction_protocol/sender.rs

```diff
@@ -72,38 +72,49 @@ pub(crate) struct OutputPair {
 /// Transaction construction process.
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 pub(super) struct RawTransactionInfo {
+    /// The transaction ID is a u64 identifier for the transaction generated by the sender
     pub tx_id: TxId,
-    // the recipient's data
+    /// Details for the construction of the recipient output. OutputFeatures etc.
     pub recipient_data: Option<RecipientDetails>,
+    /// The TransactionOutput received from the recipient.
     pub recipient_output: Option<TransactionOutput>,
+    /// The partial kernel excess received from the recipient.
     pub recipient_partial_kernel_excess: PublicKey,
+    /// The partial kernel signature received from the recipient.
     pub recipient_partial_kernel_signature: Signature,
+    /// The partial kernel offset received from the recipient.
     pub recipient_partial_kernel_offset: PrivateKey,
-    // The sender's portion of the public commitment nonce
+    /// The change output details. This may be None if no change is required.
```

Comment:
```suggestion
    /// The change output details. Always required
```

Reply:
In that case, I'll have to see if I can remove the option

Reply:
On build it will throw an error

Reply:
That doesnt appear to be true. See `try_build_transaction` and `validate`

Reply:
You are correct, dont know why I got that idea in my head.